disco_aws_automation/disco_aws.py

```diff
@@ -355,7 +356,8 @@ def provision(self, ami, hostclass=None,
 
         group = self.autoscale.get_group(
             hostclass=hostclass, launch_config=launch_config.name,
-            vpc_zone_id=",".join([subnet.id for subnet in self.get_subnets(meta_network, hostclass)]),
+            vpc_zone_id=",".join([subnet['SubnetId'] for subnet
+                                  in self.get_subnets(meta_network, hostclass)]),
             min_size=size_as_minimum_int_or_none(min_size),
             max_size=size_as_maximum_int_or_none(max_size),
             desired_size=size_as_maximum_int_or_none(desired_size),
```

Comment:
+1 on splitting some stuff out, and a lot of networking magic seems like a good candidate. Obviously, not necessarily part of this PR.